apis/otpapi/otpapi.yaml

```diff
@@ -0,0 +1,407 @@
+      requestBody:
+        description: |-
+          The body refers to the format of the request.
+            - The body contains metadata about the notification to be sent.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/generateOtpRequest'
+        required: true
+      responses:
+        200:
+          description: OK ! Successful operation."Notifying User" operation was successfully
+            executed.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ContentCreateResponse'
+              example:
+                id: api.otp.generate
+                ver: v1
+                ts: 2020-12-01 10:49:01:948+0000
+                params:
+                  resmsgid: 
+                  msgid: 12eb0ca45ce4c776fab994d120bedd41
+                  err: null
+                  status: success
+                  errmsg: null.
+                responseCode: OK
+                result: 
+                 response: "SUCCESS"
+        400:
+          description: BAD REQUEST. The "Notifying User" operation failed ! The possible
+            reason for failure is that you may have missed providing input for a mandatory
+            parameter.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ContentCreateErrorResponse'
+              example:
+                id: api.otp.generate
+                ver: v1
+                ts: 2020-12-01 10:49:01:948+0000
+                params:
+                  resmsgid: 
+                  msgid: 12eb0ca45ce4c776fab994d120bedd41
+                  err: MANDATORY_PARAMETER_MISSING
+                  status: MANDATORY_PARAMETER_MISSING
+                  errmsg: Mandatory parameter key is missing.
+                responseCode: SERVER_ERROR
+                result: {}
+        500:
+          description: INTERNAL SERVER ERROR ! Looks like something went wrong! These
+            errors are tracked automatically, but if the problem persists feel free
+            to contact us. In the meantime, try refreshing.
+          content: 
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ContentCreateErrorResponse'
+              example:
+                id: api.otp.generate
+                ver: v1
+                ts: 2020-12-01 10:49:01:948+0000
+                params:
+                  resmsgid: 
+                  msgid: 12eb0ca45ce4c776fab994d120bedd41
+                  err: INTERNAL_ERROR
+                  status: INTERNAL_ERROR
+                  errmsg: Process failed,please try again later.
+                responseCode: SERVER_ERROR
+                result: {}
+  /otp/v1/verify:
+    post:
+      tags:
+      - OTP APIs
+      summary: Verify OTP
+      description: |-
+        This API is associated with verifying OTP to user on the Sunbird Platform.
```

Comment:
@sknirmalkar89 does the API verify the OTP 'to' the user or 'of' the user? meaning does this API send a message to the user saying 'OTP verified' or does the API verify if the OTP entered by the user is correct? Please clarify and correct accordingly.

Reply:
updated